File: src/main/java/spring/springserver/domain/community/data/request/CreatePostRequest.java
package spring.springserver.domain.community.data.request;

import jakarta.validation.constraints.NotBlank;
import spring.springserver.domain.community.entity.CommunityPost;
import spring.springserver.domain.member.entity.Member;

public record CreatePostRequest(

        @NotBlank
        String title,

        String content,

        String imageUrl
) {

    public CommunityPost toEntity(Member member) {

        return CommunityPost.builder()
                .member(member)
                .username(member.getUsername())
                .title(title)
                .content(content)
                .imageUrl(imageUrl)
                .viewCount(0)
                .build();
    }
}

File: src/main/java/spring/springserver/domain/community/data/request/CreatePostRequest.kt
package spring.springserver.domain.community.data.request

import jakarta.validation.constraints.NotBlank
import spring.springserver.domain.community.entity.CommunityPost
import spring.springserver.domain.member.entity.Member

data class CreatePostRequest(
    @field:NotBlank
    val title: String,
    val content: String?,
    val fileUrl: String?,
) {
    fun toEntity(member: Member): CommunityPost {
        return CommunityPost(
            member = member,
            username = member.username,
            title = title.trim(),
            content = content?.trim()?.takeIf { it.isNotBlank() },
            fileUrl = fileUrl?.trim()?.takeIf { it.isNotBlank() },
            viewCount = 0,
            isEdited = false,
        )
    }
}
